### DISCUSSION TITLE
Adding an expenses custom field to an invoice

### DISCUSSION THREAD

**Discussion by mcfly3001** [kimai/Paid-plugins#71] [Questions & Answers] [answered]
I am currently trying to adjust the PDF invoicing template for my needs. In my case I would like to create an invoice based on an expense entry. For the expense entry I also added a custom field with the internal name "position" where I enter the position of the order I want to invoice.
Accessing this custom field though does not seem to work. For creating the invoice, I filter for the projects activity where I add my expenses. Then I am trying to access the "position" as described in the docu:

```
{% if entry['entry.meta.position'] is defined %}
    Pos: {{ entry['entry.meta.position'] }}
{% else %}
    No position found.
{% endif %}
```

This does not seem to work though. For testing, I am using the debug template which is provided in the docu. I put the code snipped above in the for-loop which iterates through all `entries` (but only uses the first one then). It seems, that `entry[entry.meta]` is not available at all. Could it be, that meta fields are only available for `timesheet` types but not for `expenses`?

**Comment by kevinpapst** [comment, marked as answer]
No to your questions.

Are you using the "standard" sum calculation? If entries are grouped, meta-fields are not supported.

**Reply by mcfly3001** [reply]
Thanks again! I did not understand that expenses will also be listed when chosing "Standard". That solved my problem.

**Reply by kevinpapst** [reply]
Yeah, I guess I need to re-word the field "sum calculation" to "group entries by".
And "standard" to "none" and the other ones accordingly.

**Reply by mcfly3001** [reply]
Rewording might make sense for a better understanding, that's right. The custom field now works as expected. Only took a bit to understand that I need to iterate the `entries` instead of the `model.calculator.entries` and that I need to access the values via the not-so-nice `entry['entry.theAttribute']` syntax.
But in the end I could find all the details in your documentation, great! :-)

**Reply by kevinpapst** [reply]
It makes more sense in DOCX templates, where it is called `${entry.theAttribute}`.

It is called like that since years and can't be changed for BC reasons.
